### ISSUE TITLE
Update order by property schema to allow precedence, and make consistent throughout the configuration objects

### ISSUE THREAD

**Issue by wwqrd** [codaco/Architect#108]
Current schema does not allow rule precedence and looks like this:
```
// since property orders should not be expected to maintain order, this could sort by name or age first
{
  'name': 'ASC',
  'age': 'DESC',
}
```

By updating schema to use arrays we can add precedence rules, by using fixed attributes we make it easier to parse:

```
[
  {
    property: 'name',
    direction: 'asc',  // capitalisation has been changed to match common libraries `sorty`/`lodash`
  },
  {
    property: 'age',
    direction: 'desc',
  },
]
```

**Comment by wwqrd** [comment]
Resolved by #111

**Comment by wwqrd** [comment]
And #555
